### ISSUE TITLE
Test results BEP events are delayed when using `coverage` and Skymeld is enabled

### ISSUE THREAD

**Issue by brentleyjones** [bazelbuild/bazel#21475]
### Description of the bug:

In Bazel 7, when using `--experimental_merged_skyframe_analysis_execution` (the default), and using `bazel coverage` (but not `bazel test`), test results are delayed until the end of the build. If using `--noexperimental_merged_skyframe_analysis_execution`, or using `bazel test`, test results appear as soon as the tests are done running.

### Which category does this issue belong to?

Core

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```shellsession
$ git checkout git@github.com:bazelbuild/bazel.git
$ cd bazel
$ USE_BAZEL_VERSION=7.0.2 bazel coverage --bes_results_url=https://app.buildbuddy.io/invocation/ --bes_backend=grpcs://remote.buildbuddy.io --cache_test_results=no //src/test/java/com/google/devtools/build/lib/...
$ USE_BAZEL_VERSION=7.0.2 bazel clean
$ USE_BAZEL_VERSION=7.0.2 bazel coverage --bes_results_url=https://app.buildbuddy.io/invocation/ --bes_backend=grpcs://remote.buildbuddy.io --cache_test_results=no --noexperimental_merged_skyframe_analysis_execution //src/test/java/com/google/devtools/build/lib/...
```

In the first run, `testResult` events aren't published until the end of the build. In the second run they are published as soon as they are done running.

### Which operating system are you running Bazel on?

macOS

### What is the output of `bazel info release`?

release 7.0.2

### Have you found anything relevant by searching the web?

I saw the message in https://github.com/bazelbuild/bazel/commit/37247d53dc828ad9d03cf4a1537d1db5d83caf99, and it seems that there was a change regarding how the combined report was generated. Maybe that is delaying these related events?

**Comment by fmeum** [comment]
Cc @joeleba

**Comment by joeleba** [comment]
@brentleyjones thanks for filing the bug. This is because of https://github.com/bazelbuild/bazel/commit/c2a6f0c38b75743158a3a8cbe7d6adf4202c2110.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.2.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.2.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.2.0rc1. Thanks!
